### PR TITLE
[GLUTEN-2756] Add a contributing guide

### PR DESCRIPTION
**Pull request by PHILO-HE**
Introduce some conventions of the community on PR, issue, coding, code review, etc., which can also reduce review cost.

### CODE REVIEW

**Comment by PHILO-HE** [comment]
@zhouyuan, @baibaichen, @ulysses-you, please spare some time to take a review. Thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  `[GLUTEN-<issue ID>][CH] xxxx`
+
+* PR just for VL backend:
+
+  `[GLUTEN-<issue ID>][VL] xxxx`
+
+* PR for common module:
+
+  `[GLUTEN-<issue ID>] xxxx`
```
Comment (winningsix) [inline CONTRIBUTING.md:26]: Just to clarify. Does this mean there is no tab for ```[CORE]``` moving forward? Also, will Github issue a must?

Comment (ulysses-you) [reply CONTRIBUTING.md:26]: I think it would be better to add `[CORE]` to mention if the code change affect both Velox and CH.

Comment (ulysses-you) [reply CONTRIBUTING.md:26]: > Also, will Github issue a must

I think a issue is required for several cases:
1. we are going to send a batch of prs for one topic, means it's a umbrella
2. a bug report or feature improvement that the reporter won't send pr directly or eagerly

Then for others, it is optional. But the description is required, developers should put something into pr description to make pr easy to follow.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+##### ClickHouse Backend CI
+To check CI failure for CH backend, please log in with the public account/password provided [here](https://github.com/oap-project/gluten/blob/main/docs/get-started/ClickHouse.md#new-ci-system).
+To re-trigger CH CI, please post the below comment on PR page:
+
+`Run Gluten Clickhouse CI`
+
+##### Velox Backend CI
+To check CI failure for Velox backend, please go into the GitHub action page from your PR page.
+
+To see the perf. impact on Velox backend, you can comment `/Benchmark Velox` on PR page to trigger a pretest. The benchmark
```
Comment (ulysses-you) [inline CONTRIBUTING.md:77]: this is cool, I have not used it. Will try it.

File: CONTRIBUTING.md (hunk 1 above)
Comment (PHILO-HE) [reply CONTRIBUTING.md]: Yes, github issue is not mandatory. `if necessary` is used in the above statement.

**Review by ulysses-you — APPROVED** [review]

**Review by FelixYBW — APPROVED** [review]

**Review by binmahone — COMMENTED** [review]
LGTM

**Comment by PHILO-HE** [comment]
Thanks everyone for the review! Will merge the doc soon.

**Pull request merged**